src/plugins/intel_cpu/src/nodes/istft.cpp

```diff
@@ -82,19 +85,163 @@ bool ISTFT::created() const {
     return getType() == Type::ISTFT;
 }
 
+namespace {
+static void transpose_out4d(const uint8_t* in,
+                            uint8_t* out,
+                            const VectorDims& in_shape,
+                            const VectorDims& out_shape,
+                            size_t elem_size) {
+    const std::vector<size_t> axes_order{0, 2, 1, 3};
+    parallel_for3d(out_shape[0],
+                   out_shape[1],
+                   out_shape[2],
+                   [in, out, axes_order, &in_shape, &out_shape, elem_size](size_t i, size_t j, size_t k) {
+                       size_t in_indexes[3];
+                       in_indexes[axes_order[0]] = i;
+                       in_indexes[axes_order[1]] = j;
+                       in_indexes[axes_order[2]] = k;
+                       size_t in_off =
+                           ((in_indexes[0] * in_shape[1] + in_indexes[1]) * in_shape[2] + in_indexes[2]) * in_shape[3];
+                       size_t out_off = ((i * out_shape[1] + j) * out_shape[2] + k) * out_shape[3];
+                       cpu_memcpy(out + out_off * elem_size, in + in_off * elem_size, out_shape[3] * elem_size);
+                   });
+}
+
+void istft_impl(const float* in_data,
+                const float* window,
+                float* final_result,
+                const ov::Shape& data_shape,
+                const ov::Shape& window_shape,
+                const int64_t frame_size,
+                const int64_t frame_step,
+                const int64_t length,
+                const bool center,
+                const bool normalized,
+                std::shared_ptr<RDFTExecutor> rdft_executor) {
+    const auto is_data_3D = data_shape.size() == 3;
+    const size_t frames_axis = 1 + (is_data_3D ? 0 : 1);
+    const size_t batch_size = is_data_3D ? 1 : data_shape[0];
+
+    const auto sqrt_frame_size = static_cast<float>(std::sqrt(frame_size));
+    const auto num_frames = data_shape[frames_axis];
+
+    const auto signal_length = (num_frames - 1) * frame_step + frame_size;
+    const int64_t final_signal_length =
+        length > 0 ? length : (center ? (signal_length - (frame_size & ~1)) : signal_length);
+    std::fill(final_result, final_result + batch_size * final_signal_length, 0.f);
+
+    std::vector<float> mid_result(batch_size * signal_length, 0.f);
+
+    const auto fft_results_dim = data_shape[data_shape.size() - 3];
+    OPENVINO_ASSERT(fft_results_dim == static_cast<size_t>((frame_size / 2) + 1));
+
+    const auto frame_size_dim = static_cast<size_t>(frame_size);
+    const auto frame_size_dim_shape = ov::Shape{frame_size_dim};
+    const auto frame_size_dim_shape_out = ov::Shape{frame_size_dim, 2};
```

Comment:
```suggestion
```
Constants `frame_size_dim_shape` and `frame_size_dim_shape_out` are not used

Reply:
Thanks, noted! I thought there are checks for unused variables:)
I will apply as a follow up.